src/ch00-15-contract-testing.md

```diff
@@ -0,0 +1,53 @@
+# Contract Testing
+
+Testing plays a crucial role in software development, especially for smart contracts. In this section, we'll guide you through the basics of testing a smart contract on Starknet with `scarb`.
+
+Let's start with a simple smart contract as an example:
+```rust
+{{#include ../listings/ch00-introduction/testing/src/lib.cairo:contract}}
+```
+
+Now, take a look at the tests for this contract:
+```rust
+{{#include ../listings/ch00-introduction/testing/src/lib.cairo:test}}
+```
+
+To define our test, we use scarb, which allows us to create a separate module guarded with `#[cfg(test)]`. This ensures that the test module is only compiled when running tests using `scarb test`.
+
+Each test is defined as a function with the `#[test]` attribute. You can also check if a test panics using the `#[should_panic]` attribute.
+
+As we are in the context of a smart contract, it's essential to set up the gas limit. You do this by using the `#[available_gas(X)]` attribute to specify the gas limit for a test. This is also a great way to ensure that your contract's features stay under a certain gas limit!
```

Comment:
> Note: The term "gas" here refers to Sierra gas, not L1 gas

Reply:
Thanks! I'll add this information as it can create confusion.